pkg/kv/kvpb/api.proto

```diff
@@ -71,6 +71,27 @@ enum RoutingPolicy {
   NEAREST = 1;
 }
 
+// ConnectionClass specifies the preferred RPC connection class to use for
+// sending or retransmitting the BatchRequest.
+//
+// This enum is similar and related to rpc.ConnectionClass, but the semantics
+// differ. This enum only defines classes used for BatchRequest traffic, and
+// only defines preferences. There are more classes in rpc package that are not
+// relevant here, such as RaftClass.
+enum ConnectionClass {
```

Comment:
Can we replace `rpc.ConnectionClass` with this enum? It seems unfortunate to have two separate enums that mean the same thing and need to be kept in sync.

Reply:
Agree, but they are not quite the same thing. Logically, this enum is neither a subset nor a superset of `rpc.ConnectionClass` (even though at the moment it is). Generally, it is a "tag".

Consider a future in which we may categorize more types of streams/traffic (schema changes, imports/exports, etc). We don't necessarily want to have that many RPC classes - some of them can be multiplexed to "bulk" class or alike. In some more distant future with QUIC, these tags can translate to separate streams more fearlessly.

Such tags are also useful for other stuff, like observability/monitoring. If we have 10 tags, we can break down the "bytes/s" graphs into 10 components. Even though all 10 can be flowing through the same class / TCP connection.

This `enum` is part of the `BatchRequest` API (which goes into the wire too), so is limited to connection classes that are used for `Send`s (and e.g. does not include `RaftClass`). It basically virtualizes away the lower-level RPC stuff - it tells the "kind" of traffic, and RPC layer chooses the physical implementation for it.

I'm not super inclined to reverse the dependency here and say that RPC depends on protos in `kvpb`. But maybe putting this into some `rpcpb` would be fine for now.

Reply:
We can build something like that if and when we need it. Let's just keep this simple and have a single concept of connection classes instead of adding yet another layer of indirection (aka complexity) that we may or may not need at some unknown point in the future.

>I'm not super inclined to reverse the dependency here and say that RPC depends on protos in kvpb.

Why not? All sorts of things depend on `kvpb` or `roachpb`. The only real difference is that they can be used in Protobufs.

Either way, let's just get this wrapped up. I'll trust your judgement on how to structure the enums.

Reply:
Done. Moved `ConnectionClass` to a new `rpcpb` package, and left aliases for the old type and constants, so that we don't need to touch many tens of usages in this PR. Added a TODO to get rid of the aliases later.